libraries/abstractions/ble_hal/test/src/iot_test_ble_hal_integration.c

```diff
@@ -219,11 +300,63 @@ TEST( Full_BLE_Integration_Test, BLE_Write_Notification_Size_Greater_Than_MTU_3
     }
 }
 
+TEST( Full_BLE_Integration_Test_Connection, BLE_Send_Data_After_Disconected )
+{
+    BTStatus_t xStatus;
+    BLETESTindicateCallback_t xIndicateEvent;
+    BLETESTInitDeinitCallback_t xInitDeinitCb;
+    BLETESTwriteAttrCallback_t xWriteEvent;
+    BLETESTreadAttrCallback_t xReadEvent;
+
+    /* Check communication */
+    xWriteEvent = prvWriteReceive( bletestATTR_SRVCB_CHAR_A, true, false, 0 );
+    prvWriteResponse( bletestATTR_SRVCB_CHAR_A, xWriteEvent, true );
+    xReadEvent = prvReadReceive( bletestATTR_SRVCB_CHAR_A );
+    prvReadResponse( bletestATTR_SRVCB_CHAR_A, xReadEvent, true );
+
+    checkNotificationIndication( bletestATTR_SRVCB_CCCD_E, true );
+    checkNotificationIndication( bletestATTR_SRVCB_CCCD_F, true );
+    prvCheckNotification( true );
+    prvCheckIndication( true );
+
+    /* Disconnect from RPi and Check communication*/
+    prvWaitConnection( false );
+
+    prvWriteResponse( bletestATTR_SRVCB_CHAR_A, xWriteEvent, false );
+    prvReadResponse( bletestATTR_SRVCB_CHAR_A, xReadEvent, false );
+    prvCheckNotification( false );
+    prvCheckIndication( false );
+
+    /* Advertise and Reconnect */
+    prvSetAdvProperty();
```

Comment:
Do we need to test reconnect again? As reconnect logic is already tested in other tests. It will save test time/